homeassistant/components/squeezebox/config_flow.py

```diff
@@ -0,0 +1,254 @@
+"""Config flow for Logitech Squeezebox integration."""
+import asyncio
+import logging
+
+from pysqueezebox import Server, async_discover
+import voluptuous as vol
+
+from homeassistant import config_entries, core, data_entry_flow, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    HTTP_UNAUTHORIZED,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+# pylint: disable=unused-import
+from .const import DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_USERNAME): str,
+        vol.Optional(CONF_PASSWORD): str,
+    }
+)
+
+TIMEOUT = 5
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """
+    Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    server = Server(
+        async_get_clientsession(hass),
+        data[CONF_HOST],
+        data[CONF_PORT],
+        data.get(CONF_USERNAME),
+        data.get(CONF_PASSWORD),
+    )
+
+    status = await server.async_query("serverstatus")
+    if not status:
+        if server.http_status == HTTP_UNAUTHORIZED:
+            raise InvalidAuth
+        raise CannotConnect
+
+    return status
+
+
+class SqueezeboxConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Logitech Squeezebox."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize an instance of the squeezebox config flow."""
+        self.data_schema = DATA_SCHEMA
+        self.discovery_info = None
+
+    async def _discover(self, uuid=None):
+        """
+        Discover an unconfigured LMS server.
+
+        Parameters:
+            uuid: search for this uuid (optional)
+        """
+        self.discovery_info = None
+
+        def _discovery_callback(server):
+            if server.uuid:
+                if uuid:
+                    # ignore non-matching uuid
+                    if server.uuid != uuid:
+                        return
+                else:
+                    # ignore already configured uuids
+                    for entry in self._async_current_entries():
+                        if entry.unique_id == server.uuid:
+                            return
+                self.discovery_info = {
+                    CONF_HOST: server.host,
+                    CONF_PORT: server.port,
+                    "uuid": server.uuid,
+                }
+                _LOGGER.debug("Discovered server: %s", self.discovery_info)
+
+        discovery_task = self.hass.async_create_task(
+            async_discover(_discovery_callback)
+        )
+        while not self.discovery_info:
+            await asyncio.sleep(0)
```

Comment:
`_discovery_callback` should set an event, and we should await the event with a timeout instead.

Reply:
Got it. Fixed in latest push.